### ISSUE TITLE
Frequent crashes when starting from terminal

### ISSUE THREAD

**Issue by dokterbob** [gitx/gitx#346]
```sh
% gitx .
2022-10-08 11:11:20.867 gitx[21413:5927913] warning: failed to get scripting definition from /Applications/GitX.app/Contents/Frameworks/Sparkle.framework/Versions/B/Updater.app; it may not be scriptable.
2022-10-08 11:11:20.867 gitx[21413:5927913] -[SBApplication open:withOptions:]: unrecognized selector sent to instance 0x7fd837612810
2022-10-08 11:11:20.867 gitx[21413:5927913] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[SBApplication open:withOptions:]: unrecognized selector sent to instance 0x7fd837612810'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff30386a17 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff68d0ba9e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff30405e36 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007fff302eb190 ___forwarding___ + 1427
	4   CoreFoundation                      0x00007fff302eab68 _CF_forwarding_prep_0 + 120
	5   gitx                                0x0000000105338689 gitx + 5769
	6   gitx                                0x00000001053395e9 gitx + 9705
	7   libdyld.dylib                       0x00007fff69eabcc9 start + 1
	8   ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException
zsh: abort      gitx .
```

GitX: 0.23 (0.23.2878)
macOS: 10.15.7 (19H2026)

This problem did not occur in GitX 0.21

**Comment by hannesa2** [comment]
Did you tried "Enable Terminal Usage" again ?
<img width="259" alt="image" src="https://user-images.githubusercontent.com/3314607/194702840-824a77fa-d7ec-4c5b-8691-7c849231e2d6.png">

**Comment by Agowan** [comment]
I can confirm the same behaviour. I did "Enable Terminal Usage" but the issue remains.
Although, I can use gitx in the terminal when gitx is already running.

It started to occur after I installed last update.

**Comment by tjmcewan** [comment]
Same here.  Gitx 0.23, macOS 12.6.  My crashes aren't "frequent", rather "every time without fail".  If the app is already open, switching to it works successfully, if not it will crash guaranteed.  I have tried replacing the binary several times.

**Comment by hannesa2** [comment]
> This looks to be a new issue with the latest ver. It happens when gitx.app is not running.

I can't reproduce it with `1.0` (the `0.22` comes with this issue, even when you installed `0.22` just only once)

Some solved it with https://github.com/gitx/gitx/issues/397#issuecomment-1503413204

**Comment by ivanoblomov** [comment]
You can avoid this error entirely by using this alias instead:
```bash
gx='open -a GitX .'
```

**Comment by goneng** [comment]
Fixed by #560.

The Sparkle warning in your log was a symptom rather than the cause. A workspace-wide `PRODUCT_BUNDLE_IDENTIFIER` override meant Sparkle's embedded `Updater.app` also claimed `net.phere.GitX`. The `gitx` tool asked LaunchServices for that identifier, was handed a bundle carrying no GitX scripting definition, and every scripting call then failed exactly as you saw:

```
-[SBApplication open:withOptions:]: unrecognized selector
```

Fixed at both ends: the override is gone, so releases stop shipping a second bundle with that identifier, and the tool now locates GitX relative to its own path instead of looking the identifier up, falling back to the old lookup only when it is run from outside an app bundle. That second half also makes the CLI work against the broken bundles that earlier releases already installed.
